torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -646,6 +765,37 @@ c10::optional<std::vector<int64_t>> GetValueFromListConstructNode(
       : c10::nullopt;
 }
 
+void SetShapeValueFromListConstructNode(Node* lc_node) {
+  auto rank = lc_node->inputs().size();
+  std::vector<c10::ShapeSymbol> shape_size;
+  for (const auto i : c10::irange(rank)) {
```

Comment:
Can you loop over the inputs directly?
`for (const auto& input : lc_node->inputs()) {`

If so you can delete the rank variable above.

Reply:
(1) loop inputs directly.
(2) We need `rank` to compare with `shape_size.size()`